### PR TITLE
A workable stan.sh for k3s single node deployment

### PR DESCRIPTION
**Pull request by frankwg**
# Why ?
We want to have only one deployment model to reduce our overall maintenance effort. K3s provides us the opportunity for single node k8s deployment, replacing the current docker based approach.

Single node deployment helps development, unit testing, also our support staffs reproducing customer's problem for better serviceability.

With proper design and arrangement, our customers can expand to multiple node deployment without much migration effort.

# What?
There are three tiers (k3s, db and instana) of deployment.
You can start with
```console
# Installing docker and k3s
./stan.sh up k3s 

# Installing instana console and the datastore layer
./stan.sh up db

# Installing instana kubectl plugin and pods of Core and Unit
./stan.sh up instana
```
all of these instructions can take an optional parameter to specific which version to be used.

You can use
```console
./stan.sh down instana
./stan.sh down db
./stan.sh down k3s
```
to restore your machine back to it original state.

### CODE REVIEW

**Comment by frankwg** [comment]
This PR should resolve the issue https://github.com/instana/lab-self-hosting-k8s/issues/8

**Review by unguiculus — CHANGES_REQUESTED** [review]
I only did a quick review. I haven't really looked at the script yet.

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,5 @@
 .project
 .vscode
 .DS_Store
+.work
+.cache
```
[the remark this answers is not in the document]
Comment (frankwg) [reply .gitignore]: This has been fixed in the commit ae4d1c285363468567fc17cbd74f3fb4b7ae234a.

File: k3s-install/conf/ambassador-service.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  externalTrafficPolicy: Local
+  ports:
+   - name: http
+     port: 80
+     protocol: TCP
+     targetPort: 8080
+   - name: https
+     port: 443
+     protocol: TCP
+     targetPort: 8443
```
[the remark this answers is not in the document]
Comment (frankwg) [reply k3s-install/conf/ambassador-service.yaml]: This has been fixed in the commit ae4d1c285363468567fc17cbd74f3fb4b7ae234a.

File: k3s-install/conf/ambassador-rbac.yaml [hunk 3]
```diff
@@ -0,0 +1,418 @@
+# GENERATED FILE: edits made by hand will not be preserved.
```
[the remark this answers is not in the document]
Comment (frankwg) [reply k3s-install/conf/ambassador-rbac.yaml]: Actually, I don't know how to use Traefik (sweat...).  There is no good document or example to follow through. Ambassador is way much easier to use in this example https://rancher.com/blog/2020/deploy-an-ingress-controllers. Maybe, I will change it to use Traefik in the future, if time permitted.

**Comment by morningspace** [comment]
I think there is still one thing may need to be clarified. Ideally, from user's perspective, they may not need to know the details about which Kubernetes distribution is being used underneath for this Instana sandbox. No matter it is k3s or kind, the only thing that user cares about is to launch a small sandbox for Instana quickly. With that, I think maybe we should combine this into the kind-based one, instead of creating two different folders to avoid confusion.

**Pull request merged**